### PR TITLE
oracleobjectstorage: Use rclone's rate limiter in multipart transfers

### PR DESCRIPTION
**Pull request by msays2000**
#### What is the purpose of this change?
In OCI-SDK the multi part uploader doesn't take rclone's pacer to rate limit the part upload calls.
Also it doesn't have throttling to not read from source if the parts upload to destination is going slow. 
Fixes:
- rewrote and made multipart upload use rclone's pacer
- Added --attempt-to-resume-upload from a past ongoing multipart upload. This is helpful to skip parts already uploaded when rclone process restarts for any reason.
- Fail fast on a part failure.
- throttle the source reading parts if destination part push is throttled.
- use memory pool map for part buffers like s3 mutipart upload.
- Added a tutorial on how to mount object storage buckets as local file system and via NFS server.

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [X] I have added tests for all changes in this PR if appropriate.
- [X] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by msays2000** [comment]
Attaching results of test_all --backends oracleobjectstorage

[TestOracleObjectStorage-2023-07-31-161049.zip](https://github.com/rclone/rclone/files/12222899/TestOracleObjectStorage-2023-07-31-161049.zip)

```
mkghosh@mkghosh-mac rclone % test_all --backends oracleobjectstorage
2023/07/31 16:10:49 Testing remotes: TestOracleObjectStorage:
2023/07/31 16:10:49 fs/sync: Making test binary "sync.test"
2023/07/31 16:10:55 fs/operations: Making test binary "operations.test"
2023/07/31 16:11:01 vfs: Making test binary "vfs.test"
2023/07/31 16:11:07 Starting run with log "oracleobjectstorage-fs.operations-TestOracleObjectStorage-1.txt"
2023/07/31 16:11:07 "./operations.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage:" - Starting (try 1/5)
2023/07/31 16:11:07 Starting run with log "oracleobjectstorage-fs.sync-TestOracleObjectStorage-1.txt"
2023/07/31 16:11:07 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage:" - Starting (try 1/5)
2023/07/31 16:11:07 Starting run with log "oracleobjectstorage-vfs-TestOracleObjectStorage-1.txt"
2023/07/31 16:11:07 Starting run with log "oracleobjectstorage-backend.oracleobjectstorage-TestOracleObjectStorage-1.txt"
2023/07/31 16:11:07 "go test -v -timeout 1h0m0s -remote TestOracleObjectStorage:" - Starting (try 1/5)
2023/07/31 16:11:07 Starting run with log "oracleobjectstorage-fs.operations-TestOracleObjectStorage-fastlist-1.txt"
2023/07/31 16:11:07 "./operations.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list" - Starting (try 1/5)
2023/07/31 16:11:07 Starting run with log "oracleobjectstorage-fs.sync-TestOracleObjectStorage-fastlist-1.txt"
2023/07/31 16:11:07 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list" - Starting (try 1/5)
2023/07/31 16:11:07 "./vfs.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage:" - Starting (try 1/5)
 2023/07/31 16:14:29 "./vfs.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage:" - Finished OK in 3m21.731913625s (try 1/5)
2023/07/31 16:16:58 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list" - Finished ERROR in 5m50.993771625s (try 1/5): exit status 1: Failed [TestServerSideCopy TestSyncOverlapWithFilter]
2023/07/31 16:16:58 Starting run with log "oracleobjectstorage-fs.sync-TestOracleObjectStorage-fastlist-2.txt"
2023/07/31 16:16:58 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list -test.run '^(TestServerSideCopy|TestSyncOverlapWithFilter)$'" - Starting (try 2/5)
2023/07/31 16:17:06 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage:" - Finished OK in 5m58.922132916s (try 1/5)
2023/07/31 16:17:13 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list -test.run '^(TestServerSideCopy|TestSyncOverlapWithFilter)$'" - Finished ERROR in 14.310506542s (try 2/5): exit status 1: Failed [TestSyncOverlapWithFilter]
2023/07/31 16:17:13 Starting run with log "oracleobjectstorage-fs.sync-TestOracleObjectStorage-fastlist-3.txt"
2023/07/31 16:17:13 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list -test.run '^TestSyncOverlapWithFilter$'" - Starting (try 3/5)
2023/07/31 16:17:21 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list -test.run '^TestSyncOverlapWithFilter$'" - Finished ERROR in 8.053314041s (try 3/5): exit status 1: Failed [TestSyncOverlapWithFilter]
2023/07/31 16:17:21 Starting run with log "oracleobjectstorage-fs.sync-TestOracleObjectStorage-fastlist-4.txt"
2023/07/31 16:17:21 "./sync.test -test.v -test.timeout 1h0m0s -remote TestOracleObjectStorage: -fast-list -test.run '^TestSyncOverlapWithFilter$'" - Starting (try 4/5)
2023/07/31 16:17:26 "go test -v -timeout 1h0m0s -remote TestOracleObjectStorage:" - Found 5 ignored errors only - marking as good
2023/07/31 16:17:26 ------------------------------------------------------------

```

**Comment by msays2000** [comment]
Screen Shot 2023-07-31 at 4 28 14 PM

**Comment by msays2000** [comment]
@ncw kindly review thank you!

File: docs/content/oracleobjectstorage.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ versionIntroduced: "v1.60"
 Paths are specified as `remote:bucket` (or `remote:` for the `lsd` command.)  You may put subdirectories in 
 too, e.g. `remote:bucket/path/to/dir`.
 
+## Tutorials
+
+- [Mount buckets as local file system](../../backend/oracleobjectstorage/docs/oos-mount.md)
```
Comment (ncw) [inline docs/content/oracleobjectstorage.md:17]: This link won't work on the website as we don't expose anything from backend.

It needs to go in docs/content

I think (and I haven't tried this!) you can turn `docs/content/oracleobjectstorage.md` into a directory `docs/content/oracleobjectstorage/index.md` then that gives you an obvious place to put the extra docs. I don't think this changes the URL of the main docs page.

**Review by ncw — COMMENTED** [review]
I have given this a quick once over and it is looking great :-)

I have a query about your additional doc - a suggested place for it is inline.

You can see what the docs should look like if you install [hugo](https://gohugo.io/) then run `make serve` and that will tell you if the new place works OK,

There is an S3 test broken at the moment - you are right. I haven't got round to fixing it yet!

File: docs/content/oracleobjectstorage.md (hunk 1 above)
Comment (msays2000) [reply docs/content/oracleobjectstorage.md]: @ncw thank you for surfacing the docs bug. I have fixed it.
- moved the tutorials to docs/content

**Comment by msays2000** [comment]
@ncw this PR is done, kindly review and merge.

**Review by ncw — APPROVED** [review]
This looks great now - thank you :-)

I'll merge it now.

**Pull request merged**